homeassistant/components/srp_energy/config_flow.py

```diff
@@ -1,65 +1,90 @@
 
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
 
-class ConfigFlow(config_entries.ConfigFlow, domain=SRP_ENERGY_DOMAIN):
-    """Handle a config flow for SRP Energy."""
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    LOGGER.debug("Validate User Input with client")
+    srp_client = SrpEnergyClient(
+        data[CONF_ID],
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+    )
 
-    VERSION = 1
+    is_valid = await hass.async_add_executor_job(srp_client.validate)
+
+    LOGGER.debug("Is user input valid: %s", is_valid)
+    if not is_valid:
+        raise InvalidAuth
+
+    return is_valid
 
-    config = {
-        vol.Required(CONF_ID): str,
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
-        vol.Optional(CONF_IS_TOU, default=False): bool,
-    }
 
-    async def async_step_user(self, user_input=None):
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an SRP Energy config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
         """Handle a flow initialized by the user."""
+        LOGGER.debug("Config entry")
         errors = {}
+        default_title: str = DEFAULT_NAME
 
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
 
-        if user_input is not None:
-            try:
-                srp_client = SrpEnergyClient(
-                    user_input[CONF_ID],
-                    user_input[CONF_USERNAME],
-                    user_input[CONF_PASSWORD],
-                )
-
-                is_valid = await self.hass.async_add_executor_job(srp_client.validate)
-
-                if is_valid:
-                    return self.async_create_entry(
-                        title=user_input[CONF_NAME], data=user_input
-                    )
-
-                errors["base"] = "invalid_auth"
+        if self.hass.config.location_name:
+            default_title = self.hass.config.location_name
 
+        if user_input:
+            try:
+                await validate_input(self.hass, user_input)
             except ValueError:
+                # Thrown when the account id is malformed
                 errors["base"] = "invalid_account"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
             except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
+                LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+            else:
+                await self.async_set_unique_id(user_input[CONF_ID])
```

Comment:
Don't add this feature now. Move it to the other PR where we allow more than one config entry.